utils/oscap-info.c

```diff
@@ -89,21 +91,103 @@ static inline void _print_xccdf_status(struct xccdf_status *status, const char *
 	}
 }
 
-static inline void _print_xccdf_profiles(struct xccdf_profile_iterator *prof_it, const char *prefix)
+static void _remove_occurence_of_character_from_string(char *string, char c)
 {
-	printf("%sProfiles:\n", prefix);
+	char *reading_ptr = string, *writing_ptr = string;
+	while (*reading_ptr) {
+		*writing_ptr = *reading_ptr;
+		if (*reading_ptr != c)
+			writing_ptr++;
+		reading_ptr++;
+	}
+	*writing_ptr = '\0';
+}
+
+static void _print_xccdf_profile_default(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *title_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(title_it, NULL);
+	oscap_text_iterator_free(title_it);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	printf("%s\tTitle: %s\n", prefix, profile_title);
+	free(profile_title);
+	printf("%s\t\tId: %s%s\n", prefix,
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+}
+
+static void _print_xccdf_profile_verbose(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *text_it;
+	struct oscap_string_iterator *str_it;
+
+	printf("%sProfile\n", prefix);
+
+	text_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	printf("%s\tTitle: %s\n", prefix, profile_title);
+	free(profile_title);
+
+	printf("%s\tId: %s%s\n", prefix,
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+
+	text_it = xccdf_profile_get_description(prof);
+	char *profile_description = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_description, '\n');
```

Comment:
Personally, I wouldn't remove '\n' characters from profile description, because it sometimes leads to merging words of description. See this example:

```
[jcerny@thinkpad openscap{feature/info-improvement}]$ myoscap info --profile xccdf_org.ssgproject.content_profile_nist-800-171-cui  /usr/share/xml/scap/ssg/content/ssg-rhel7-ds.xml 
Document type: Source Data Stream
Imported: 2017-08-29T22:29:14

Stream: scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml
Generated: (null)
Version: 1.2
Profile
        Title: Unclassified Information in Non-federal Information Systems and Organizations (NIST 800-171)
        Id: xccdf_org.ssgproject.content_profile_nist-800-171-cui
        Description: From NIST 800-171, Section 2.2:Security requirements for protecting the confidentiality of CUI in nonfederal information systems and organizations have a well-defined structure that consists of:(i) a basic security requirements section;(ii) a derived security requirements section.The basic security requirements are obtained from FIPS Publication 200, whichprovides the high-level and fundamental security requirements for federalinformation and information systems. The derived security requirements, whichsupplement the basic security requirements, are taken from the security controlsin NIST Special Publication 800-53.This profile configures Red Hat Enterprise Linux 7 to the NIST SpecialPublication 800-53 controls identified for securing Controlled UnclassifiedInformation (CUI).
        Platforms: 
[jcerny@thinkpad openscap{feature/info-improvement}]$ 
```
Notice  federalinformation, UnclassifiedInformation, SpecialPublication, whichsupplement , etc

Reply:
This is a good point, maybe it is better to substitute newline with space instead of trashing it?

Reply:
Maybe it is better to not use the _remove_occurence_from function on profile description at all instead of modyfying it? We don't need it so strongly for profile descriptions. Newline characters in profile description will not break @dahaic